src/core/thread/address_resolver.cpp

```diff
@@ -248,8 +305,8 @@ otError AddressResolver::UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAd
 
             if (mCache[i].mState != Cache::kStateCached)
             {
-                mCache[i].mRetryTimeout        = 0;
                 mCache[i].mLastTransactionTime = static_cast<uint32_t>(kLastTransactionTimeInvalid);
+                mCache[i].mRetryTimeout        = 0;
```

Comment:
One suggestion: How about if we call `MoveCacheEntryBehindCached(mChache[i])` from `UpdateCacheEntry()` when we transition the entry from query to cached state?

Reply:
Why wouldn't we call `MoveCacheEntryAtFront()`? If this entry was in `kStateQuery`, I assume that this entry is actually being used.